factcast-factus/src/main/java/org/factcast/factus/projection/WriterTokenAware.java

```diff
@@ -20,10 +20,10 @@
 import org.factcast.factus.FactusConstants;
 
 public interface WriterTokenAware {
-  default AutoCloseable acquireWriteToken() {
+  default WriterToken acquireWriteToken() {
     return acquireWriteToken(FactusConstants.FOREVER);
   }
 
   /** might return null if token cannot be acquired */
-  AutoCloseable acquireWriteToken(@NonNull Duration maxWait);
+  WriterToken acquireWriteToken(@NonNull Duration maxWait);
```

Comment:
This would be an incompatible change that might break user code, right?

How about this solution (not very beautiful, but would not break anything):

```java
default AutoCloseable acquireWriteToken() {
  return acquireWriteToken(FactusConstants.FOREVER);
}

default AutoCloseable acquireWriteToken(@NonNull Duration maxWait){
  // note change in method name, writeToken -> writerToken
  throw new RuntimeException("Please implement acquireWriterToken!");
}

// note change in method name, writeToken -> writerToken
default WriterToken acquireWriterToken() {
  return acquireNewWriteToken(FactusConstants.FOREVER);
}

default WriterToken acquireWriterToken(@NonNull Duration maxWait){
   // If this is not overridden, assume we have legacy code overriding acquireWriteToken 
  AutoCloseable token = acquireWriteToken(maxWait);
  return new WriterToken() {

      @Override
      public void close() throws Exception {
        token.close();
      }

      @Override
      public boolean isValid() {
        return true;
      }
  };

}

```

And setting the old method deprecated and remove that sometime later?

Reply:
how is replacing an interface-return by an implementation return incompatible ?

Reply:
this feels like trading a compile error for a runtime-error. where am i wrong?

Reply:
Yeah, for the sake of guaranteeing compatibility.